client/src/partials/fiscal/fiscal.create.js

```diff
@@ -1,427 +1,446 @@
-        }
-        account.depth = depth;
-      });
-    }
+  // adds acount depth into the equation
+  function parseAccountDepth(accountModel) {
+    accountModel.data.forEach(function (account) {
+      var parent, depth = 0;
+      parent = accountModel.get(account.parent);
+      while (parent) {
+        depth += 1;
+        parent = accountModel.get(parent.parent);
+      }
+      account.depth = depth;
+    });
+  }
 
-    // STEP 1: transitions module to create fiscal year details
-    function stepOne() {
-      $scope.step = steps[0];
-    }
+  // STEP 1: transitions module to create fiscal year details
+  function stepOne() {
+    $scope.step = steps[0];
+  }
 
-    // STEP 2: transitions module state to either
-    //  1) import opening balances from a previous fiscal year
-    //  2) create new opening balances
-    function stepTwo() {
-      var hasPreviousYear = angular.isDefined(data.previous_fiscal_year);
+  // STEP 2: transitions module state to either
+  //  1) import opening balances from a previous fiscal year
+  //  2) create new opening balances
+  function stepTwo() {
+    var hasPreviousYear = angular.isDefined(data.previous_fiscal_year);
+    if ($scope.data.start < $scope.data.end) {
       $scope.step = steps[hasPreviousYear ?  1 : 2];
       if (hasPreviousYear) {
         session.previous_fiscal_year = $scope.fiscal.get(data.previous_fiscal_year).fiscal_year_txt;
       }
+    } else {
+      var msg = $translate.instant('FISCAL_YEAR.DATE_ORDER_ERROR')
+      .replace('%START%', util.htmlDate($scope.data.start))
+      .replace('%END%', util.htmlDate($scope.data.end));
+      messenger.error(msg);
     }
 
-    function checkClosing (close_fy, previous_fy_id) {
-      if (close_fy) {
-        stepThree(previous_fy_id);
-      } else {
-        $scope.createWithoutClosing = true;
-        $scope.step = steps[2];
-      }
-    }
+  }
 
-    // STEP 3: opening with resultat
-    function stepThree(id) {
-      var fy = getFiscalYear(id);
-      session.previous_fiscal_year = fy.fiscal_year_txt;
-      session.previous_fiscal_year_id = fy.id;
+  function checkClosing (close_fy, previous_fy_id) {
+    if (close_fy) {
+      stepThree(previous_fy_id);
+    } else {
+      $scope.createWithoutClosing = true;
+      $scope.step = steps[2];
+    }
+  }
 
-      getSolde(6, fy.id)
-      .then(function (data6) {
-        session.array6 = data6.data;
-        return getSolde(7, fy.id);
-      })
-      .then(function (data7) {
-        session.array7 = data7.data;
-      })
-      .then(function () {
-        session.solde6 = session.array6.reduce(sumDebMinusCred, 0);
-        session.solde7 = session.array7.reduce(sumCredMinusDeb, 0);
-        observation();
+  // STEP 3: opening with resultat
+  function stepThree(id) {
+    var fy = getFiscalYear(id);
+    session.previous_fiscal_year = fy.fiscal_year_txt;
+    session.previous_fiscal_year_id = fy.id;
+
+    getSolde(6, fy.id)
```

Comment:
We should avoid the account_number user to keep the system independant to an accounting system